### PR TITLE
+doc #743 add more advanced handleNotFound example

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka-http/issues/743

### CODE REVIEW

File: docs/src/test/java/docs/http/javadsl/server/directives/ExecutionDirectivesExamplesTest.java [hunk 1]
```diff
@@ -72,4 +72,37 @@ public void testHandleRejections() {
...
+            complete(StatusCodes.NOT_FOUND, "The path " + path + " was not found!")
+          )
+        )
+        .build();
+  
+      final Route route = 
+        handleRejections(totallyMissingHandler, () ->
+        pathPrefix("handled", () ->
+          route(
+            path("existing", () -> complete("This path exists"))
```
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/server/directives/ExecutionDirectivesExamplesTest.java]: Maybe break after the `->` to keep consistency in styles?

File: docs/src/main/paradox/java/http/routing-dsl/rejections.md [hunk 2]
```diff
@@ -121,3 +121,14 @@ all cases your handler doesn't handle itself) and used for all rejections that a
 itself.
 
 The second case allows you to restrict the applicability of your handler to certain branches of your route structure.
+
+
+#### Adding the unmatched route in handleNotFound
+
+Since rejection handlers are routes themselfs, it is possible to do anything you could possibly want inside such handler.
```
Comment (jrudolph) [inline docs/src/main/paradox/java/http/routing-dsl/rejections.md]: "themselves"

**Comment by ktoso** [comment]
Thanks!

File: docs/src/test/java/docs/http/javadsl/server/directives/ExecutionDirectivesExamplesTest.java (hunk 1 above)
Comment (ktoso) [inline docs/src/test/java/docs/http/javadsl/server/directives/ExecutionDirectivesExamplesTest.java]: Forgot to apply this feedback, sorry. Will add in other PR

Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/server/directives/ExecutionDirectivesExamplesTest.java]: Not a big deal

**Pull request merged**